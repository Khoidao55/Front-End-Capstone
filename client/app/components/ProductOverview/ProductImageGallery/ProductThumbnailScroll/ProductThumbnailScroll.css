.thumbnail-photos {
  margin-bottom: 5px;
}

.thumbnail-size {
  width: 20%;
}

.thumbnail-single-photo {
  transition: transform 450ms;
}

.thumbnail-single-photo:hover {
  transform: scale(1.08);
  opacity: 1;
}

.selected-image {
  border-style: solid;
}
