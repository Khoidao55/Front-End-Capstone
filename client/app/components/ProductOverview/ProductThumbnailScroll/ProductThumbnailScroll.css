.thumbnail-photos {
  margin-bottom: 5px;
}

.thumbnail-size {
  width: 14%;
  border-style: white;
  transition: transform 250ms;
}

/* .thumbnail-single-photo {
  transition: transform 250ms;
} */

.thumbnail-size:hover {
  transform: scale(1.08);
  opacity: 1;
}

.selected-image {
  border-style: solid;
  border-color: rgb(88, 71, 100);
  width: 14%;
}

.test {
  height: 100px;
}

.empty-thumbnail {
  width: 310px;
}