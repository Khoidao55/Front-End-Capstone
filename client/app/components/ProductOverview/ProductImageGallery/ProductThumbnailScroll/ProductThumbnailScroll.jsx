import React from 'react';
import propTypes from 'proptypes';
import './ProductThumbnailScroll.css';

const ProductThumbnailScroll = ({ currentPhoto, arrayOfPhoto }) => {
  if (arrayOfPhoto !== undefined && currentPhoto !== undefined) {
    return (
      <div className="test">
        {arrayOfPhoto.photos.map((photo) => {
          return (
          <div key={photo.thumbnail_url} className="thumbnail-photos">
            <img className={`${currentPhoto === photo.thumbnail_url ? 'selected-image thumbnail-single-photo thumbnail-size' :`thumbnail-single-photo thumbnail-size`}`} src={photo.thumbnail_url} />
          </div>
          )
        })}
      </div>
    );
  }

  return (
    <div />
  );
};

ProductThumbnailScroll.propTypes = {
  currentPhoto: propTypes.string.isRequired,
  arrayOfPhoto: propTypes.object.isRequired,
};

export default ProductThumbnailScroll;
