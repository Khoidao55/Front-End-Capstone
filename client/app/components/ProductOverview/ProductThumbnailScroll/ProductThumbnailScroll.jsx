import React from 'react';
import propTypes from 'proptypes';
import './ProductThumbnailScroll.css';

const ProductThumbnailScroll = ({ onClickChangeThumbnail, currentPhoto, arrayOfPhoto }) => {
  if (arrayOfPhoto !== undefined && currentPhoto !== undefined) {
    if (arrayOfPhoto.photos.length === 1 && currentPhoto === null) {
      return (
        <div />
      );
    }

    return (
      <div className="test">
        {arrayOfPhoto.photos.map((photo) => {
          return (
          <div key={photo.thumbnail_url} className="thumbnail-photos">
            <img onClick={() => onClickChangeThumbnail(arrayOfPhoto.photos.indexOf(photo))} className={`${currentPhoto === photo.thumbnail_url ? 'selected-image  thumbnail-size' :`thumbnail-size`}`} src={photo.thumbnail_url} />
          </div>
          )
        })}
        <button>Down</button>
      </div>
    );
  }
};

ProductThumbnailScroll.propTypes = {
  onClickChangeThumbnail: propTypes.func,
  currentPhoto: propTypes.string,
  arrayOfPhoto: propTypes.object,
};

export default ProductThumbnailScroll;
