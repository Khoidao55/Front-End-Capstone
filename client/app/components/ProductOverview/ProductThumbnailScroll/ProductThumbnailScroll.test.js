import React from 'react';
import { shallow } from 'enzyme';
import ProductThumbnailScroll from './ProductThumbnailScroll.jsx';

describe('Product Thumbnail Bar', () => {
  test('should render my component', () => {
    const wrapper = shallow(<ProductThumbnailScroll />);

    expect(wrapper).toMatchSnapshot();
  });
});
