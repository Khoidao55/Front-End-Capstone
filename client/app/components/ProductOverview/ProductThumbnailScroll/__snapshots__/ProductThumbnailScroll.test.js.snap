// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product Thumbnail Bar should render my component 1`] = `ShallowWrapper {}`;
